core/domain/user_services_test.py

```diff
@@ -1124,13 +1099,11 @@ def test_legacy_user(self):
         after the last edited time check was introduced.
         """
         # Set up a 'previous-generation' user.
-        user_settings = user_services.get_user_settings(self.owner_id)
-        user_settings.last_created_an_exploration = None
-        user_services._save_user_settings(user_settings)  # pylint: disable=protected-access
-
         owner_settings = user_services.get_user_settings(self.owner_id)
         self.assertIsNone(owner_settings.last_created_an_exploration)
 
+        user_services.record_user_created_an_exploration(self.owner_id)
```

Comment:
Why is this needed? Shouldn't it happen automatically as part of save_new_valid_exploration() below (otherwise the previous version of this test would fail)?

Reply:
You are right. Removed.